src/sentry/models/group.py

```diff
@@ -626,7 +612,59 @@ def is_resolved(self):
         return self.get_status() == GroupStatus.RESOLVED
 
     def has_replays(self):
-        return get_replay_count_for_group(self) > 0
+        def make_snuba_params_for_replay_count_query():
+            return SnubaParams(
+                organization=self.project.organization,
+                projects=[self.project],
+                user=None,
+                start=datetime.now() - timedelta(days=14),
+                end=datetime.now(),
+                environments=[],
+                teams=[],
+            )
+
+        def _cache_key(issue_id):
+            return f"has_replays:{issue_id}"
+
+        from sentry.replays.usecases.replay_counts import get_replay_counts
+        from sentry.search.events.types import SnubaParams
+
+        metrics.incr("group.has_replays")
+
+        # XXX(jferg) Note that this check will preclude backend projects from receiving the "View Replays"
+        # link in their notification. This will need to be addressed in a future change.
+        if not self.project.flags.has_replays:
+            metrics.incr("group.has_replays.project_has_replays_false")
+            return False
+
+        cached_has_replays = cache.get(_cache_key(self.id))
+        if cached_has_replays is not None:
+            metrics.incr(
+                "group.has_replays_cached",
```

Comment:
Should this metric name match the others with `.has_replays.` being a portion in there?
```suggestion
                "group.has_replays.cached",
```